### PR TITLE
Including revisions

### PR DESCRIPTION
**Pull request by reebhub**

### CODE REVIEW

**Review by omerlewitz — CHANGES_REQUESTED** [review]

File: Documentation/5.2/Raven.Documentation.Pages/client-api/session/revisions/including.markdown [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  requiring additional trips to the server.  
+* Revisions can be Included by their **Creation Date** or **Change Vector**, 
+  and retrieved when documents are -  
+   * Loaded using `session.Load`  
+   * Queried using `session.Query`  
+   * Queried using `session.Advanced.RawQuery`  
+
+- [Including Revisions](../../../client-api/session/revisions/including#including-revisions)  
+- [`IncludeRevisions`](../../../client-api/session/revisions/including#includerevisions)  
+- [Including Revisions With `session.Load`](../../../client-api/session/revisions/including#including-revisions-with-session.load)
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/client-api/session/revisions/including.markdown:19]: done

File: Documentation/5.2/Raven.Documentation.Pages/client-api/session/revisions/including.markdown [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+   * Loaded using `session.Load`  
+   * Queried using `session.Query`  
+   * Queried using `session.Advanced.RawQuery`  
+
+- [Including Revisions](../../../client-api/session/revisions/including#including-revisions)  
+- [`IncludeRevisions`](../../../client-api/session/revisions/including#includerevisions)  
+- [Including Revisions With `session.Load`](../../../client-api/session/revisions/including#including-revisions-with-session.load)  
+   - [By Date](../../../client-api/session/revisions/including#load-include-revisions-by-date)  
+   - [By Change Vector](../../../client-api/session/revisions/including#load-include-revisions-by-change-vector)  
+- [Including Revisions With `session.Query`](../../../client-api/session/revisions/including#including-revisions-with-session.query)
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/client-api/session/revisions/including.markdown:22]: done

File: Documentation/5.2/Raven.Documentation.Pages/client-api/session/revisions/including.markdown [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+---
+
+#### Load: Include Revisions by Change Vector
+
+* To include a **single revision** by a document's change vector:  
+  Pass `IncludeRevisions` the path to a property of the document you load, 
+  that contains the change vector.  
+
+* To include **a Group of revisions** by their change vectors:
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/client-api/session/revisions/including.markdown:134]: you mean "change vector"? it's basically capitalized in main titles and links and uncapitalized everywhere else. that's fine.

File: Documentation/5.2/Raven.Documentation.Pages/client-api/session/revisions/including.markdown [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+- [Including Revisions](../../../client-api/session/revisions/including#including-revisions)  
+- [`IncludeRevisions`](../../../client-api/session/revisions/including#includerevisions)  
+- [Including Revisions With `session.Load`](../../../client-api/session/revisions/including#including-revisions-with-session.load)  
+   - [By Date](../../../client-api/session/revisions/including#load-include-revisions-by-date)  
+   - [By Change Vector](../../../client-api/session/revisions/including#load-include-revisions-by-change-vector)  
+- [Including Revisions With `session.Query`](../../../client-api/session/revisions/including#including-revisions-with-session.query)  
+   - [By Date](../../../client-api/session/revisions/including#query-include-revisions-by-date)  
+   - [By Change Vector](../../../client-api/session/revisions/including#query-include-revisions-by-change-vector)  
+- [Including Revisions with `session.Advanced.RawQuery`](../../../client-api/session/revisions/including#including-revisions-with-session.advanced.rawquery)  
+   - [By Date](../../../client-api/session/revisions/including#raw-query-include-revisions-by-date)
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/client-api/session/revisions/including.markdown:26]: changed to "time" or "creation time" wherever needed

File: Documentation/5.2/Raven.Documentation.Pages/client-api/session/revisions/including.markdown [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+| **path** | `Expression<Func<T, IEnumerable<string>>>` | A path to a document property that contains **an array of change vectors**. <br> The revisions whose change vectors are contained in the array will be Included. |
+
+{PANEL/}
+
+
+{PANEL: Including Revisions With `session.Load`}
+
+#### Load: Include Revisions by Date
+
+To include a **single revision** by its **creation date**, pass `IncludeRevisions` the date.
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/client-api/session/revisions/including.markdown:120]: "date" changed to "time" or "creation time" wherever needed

**Review by omerlewitz — APPROVED** [review]

File: Documentation/5.2/Raven.Documentation.Pages/client-api/how-to/handle-document-relationships.dotnet.markdown [hunk 6]
```diff
@@ -0,0 +1,293 @@
+# Client API: How to Handle Document Relationships
```
Comment (ayende) [inline Documentation/5.2/Raven.Documentation.Pages/client-api/how-to/handle-document-relationships.dotnet.markdown:1]: Why is this here? Does this relates to the include revisions PR?

File: Documentation/5.2/Raven.Documentation.Pages/client-api/session/revisions/including.markdown [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+or to a [Raw Query](../../../client-api/session/revisions/including#including-revisions-with-session.advanced.rawquery) 
+using `.AddParameter`.  
+
+* If the provided time matches the creation time of a document revision, this revision will be included.  
+* If no exact match is found, the nearest revision to precede it will be included.  
+   * E.g. -  
+     If the time you provided is `April 7 2020`,  
+     and it is located between two existing revisions dated `April 2 2020` and `April 11 2020`,  
+     the `April 2` revision will be included.  
+* If no revisions exist, an empty `IncludeRevisions` object will be returned.
```
Comment (ayende) [inline Documentation/5.2/Raven.Documentation.Pages/client-api/session/revisions/including.markdown:68]: No, we'll remember that no such revision exists and skip going to the server when you ask for it, returning `null`

File: Documentation/5.2/Raven.Documentation.Pages/client-api/how-to/handle-document-relationships.dotnet.markdown (hunk 6 above)
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/client-api/how-to/handle-document-relationships.dotnet.markdown:1]: yes, there's an "Includes" section in it that references the different kinds of includes, added revisions to the list

File: Documentation/5.2/Raven.Documentation.Pages/client-api/session/revisions/including.markdown (hunk 7 above)
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/client-api/session/revisions/including.markdown:68]: done

**Comment by ppekrol** [comment]
New PR: https://github.com/ravendb/docs/pull/1447

**Pull request closed without merging**